api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -222,19 +229,26 @@ private String getUrlPath(Job job, String filePath) {
     )
     @ResponseStatus(value = HttpStatus.OK)
     @GetMapping(value = "/Job/{jobUuid}/file/{filename}")
-    public ResponseEntity<Resource> downloadFile(
+    public void downloadFile(
             @ApiParam(value = "A job identifier", required = true) @PathVariable @NotBlank String jobUuid,
             @ApiParam(value = "A file name", required = true) @PathVariable @NotBlank String filename) throws IOException {
         MDC.put(JOB_LOG, jobUuid);
         MDC.put(FILE_LOG, filename);
         log.info("Request submitted to download file");
 
         Resource downloadResource = jobService.getResourceForJob(jobUuid, filename);
-        HttpHeaders headers = new HttpHeaders();
-        headers.set(HttpHeaders.CONTENT_TYPE, NDJSON_FIRE_CONTENT_TYPE);
+        RequestAttributes requestAttributes = RequestContextHolder.getRequestAttributes();
+        HttpServletResponse response = ((ServletRequestAttributes) requestAttributes).getResponse();
+        response.setContentType(NDJSON_FIRE_CONTENT_TYPE);
+        OutputStream out = response.getOutputStream();
+        FileInputStream in = new FileInputStream(downloadResource.getFile());
+
+        IOUtils.copy(in, out);
 
         log.info("Sending file to client");
 
-        return new ResponseEntity<>(downloadResource, headers, HttpStatus.OK);
+        out.close();
+        in.close();
+        downloadResource.getFile().delete();
```

Comment:
Let's delegate to `jobService.deleteResourceForJob(jobUuid, filename)` here. It's best to free the controller from dealing with file management etc and let the Service layer encapsulate all that logic.

`deleteResourceForJob(jobUuid, filename)` would just be a one-liner such as `this.getResourceForJob(jobUuid, filename).getFile().delete()`.

Reply:
I moved to the service with the file, I think I have to perform the delete after I've gathered the FileInputStream since it's used in copying the contents of the file.

Reply:
Correct. `deleteResourceForJob(jobUuid, filename)` would be the last line in the controller's method. It won't be reached if copying errors out.